fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/EquivalenceClasses.java

```diff
@@ -0,0 +1,57 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.sql.optimizer.base;
+
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class EquivalenceClasses {
+    private final Map<ColumnRefOperator, Set<ColumnRefOperator>> columnToEquivalenceClass;
```

Comment:
why do need to add column A  to the EquivalenceClasses of column A?

Reply:
because EquivalenceClasses will be used to rewrite predicates and output columns. I want to make each element in the same EquivalenceClasses will have the same EquivalenceClasses, then I can pick the first element of EquivalenceClasses as rewritten column.